src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2145,31 +2145,56 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
 
         public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement localFunc)
         {
-            var oldSymbol = this.CurrentSymbol;
             var localFuncSymbol = localFunc.Symbol;
-            var delegateInvokeMethod = _delegateInvokeMethod;
-            this.CurrentSymbol = localFuncSymbol;
-            _delegateInvokeMethod = null;
-
-            var oldPending = SavePending(); // we do not support branches into a lambda
-
-            var savedState = this.State;
-            var localFunctionState = GetOrCreateLocalFuncUsages(localFuncSymbol);
+            var localFunctionState = GetOrCreateLocalFuncUsages(localFunc.Symbol);
             // The starting state is the top state, but with captured
             // variables set according to Joining the state at all the
             // local function use sites
-            State = TopState();
+            var state = TopState();
             for (int slot = 1; slot < localFunctionState.StartingState.Capacity; slot++)
             {
                 var symbol = variableBySlot[RootSlot(slot)].Symbol;
                 if (Symbol.IsCaptured(symbol, localFunc.Symbol))
                 {
-                    State[slot] = localFunctionState.StartingState[slot];
+                    state[slot] = localFunctionState.StartingState[slot];
                 }
             }
             localFunctionState.Visited = true;
 
-            if (!localFunc.WasCompilerGenerated) EnterParameters(localFuncSymbol.Parameters);
+            AnalyzeLocalFunctionOrLambda(
```

Comment:
>AnalyzeLocalFunctionOrLambda [](start = 12, length = 28)

Are we saving and restoring `_returnTypesOpt`?

Reply:
Hmm, it does seem like maybe this save-restore should be pushed into `AnalyzeLocalFunctionOrLambda`. I'll try and write a test which fails with this current implementation.

Reply:
Done